### PR TITLE
Fix #19647 - Allow importing multiple tokens

### LINKED ISSUE #19647: [Bug]: Cannot Import Multiple Tokens at One Time
### Describe the bug

While working on https://github.com/MetaMask/metamask-extension/pull/19553, I noticed that I cannot import multiple tokens at once.  Only the last token in the desired list gets imported:

https://github.com/MetaMask/metamask-extension/assets/46655/8e784ed8-df33-4d19-b383-f2efaa274608

This is presently an issue outside of my PR, on `develop` branch.

### Steps to reproduce

1.  From the home screen, click "Import Tokens"
2. Search for anything
3. Select multiple tokens to import
4. Click Next, then Confirm
5. BUG: See that only token added was the last in the list

### Error messages or log output

```shell
None.  I also debugged to see calls to the background for `addToken` were properly executed for each token.
```

### Version

develop

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

* Fixes #19647

This is a long-standing bug where trying to import multiple tokens at once didn't work, only the last would be added.  This PR fixes that issue.

## Screenshots/Screencaps

https://github.com/MetaMask/metamask-extension/assets/46655/5e1b6108-dd61-4a82-a4a2-94ec90cbac06

## Manual Testing Steps

1.  From the home screen, click "Import Tokens"
2. Search for and add multiple tokens
3. See all tokens added to the home screen

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]
Amazing! I am so happy to see these changes 😍

File: test/e2e/tests/import-tokens.spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
```
Comment (DDDDDanica) [inline test/e2e/tests/import-tokens.spec.js:30]:
```suggestion
        await unlockWallet(driver)
```
:)))

File: test/e2e/tests/import-tokens.spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        // Token list is only on mainnet
+        await driver.clickElement('[data-testid="network-display"]');
+        await driver.clickElement({ text: 'Ethereum Mainnet', tag: 'button' });
+
+        // Wait for network to change and token list to load from state
+        await driver.delay(regularDelayMs);
```
Comment (DDDDDanica) [inline test/e2e/tests/import-tokens.spec.js:37]: question: there's a delay in finding elements `[data-testid="import-token-button"]` when we execute `clickElement`, so is this still necessary?

File: test/e2e/tests/import-tokens.spec.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        await driver.clickElement('[data-testid="import-token-button"]');
+        await driver.fill('input[placeholder="Search tokens"]', 'cha');
+
+        await driver.clickElement('.token-list__token');
+        await driver.clickElement('.token-list__token:nth-of-type(2)');
+        await driver.clickElement('.token-list__token:nth-of-type(3)');
```
[the remark this answers is not in the document]
Comment (darkwing) [reply test/e2e/tests/import-tokens.spec.js:44]: They're dynamic search results, so assigning a `data-testid` with its index would be just as arbitrary.

File: test/e2e/tests/import-tokens.spec.js (hunk 2 above)
Comment (darkwing) [reply test/e2e/tests/import-tokens.spec.js:37]: Having arbitrary delays is a bad thing but when I do something like the following:

```
await driver.waitForElementNotPresent('.loading-overlay');
```

...the detection is so fast that the search list isn't available yet, and so a search shows no results.

**Review by NidhiKJha — APPROVED** [review]

File: test/e2e/tests/import-tokens.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply test/e2e/tests/import-tokens.spec.js:44]: Gotcha !

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**